### PR TITLE
docs: update LOTUS_RELEASE_FLOW with recent FAQs and branch/tag strategy items

### PR DESCRIPTION
**Pull request by BigLep**
This was done as additional content to potentially add for https://github.com/filecoin-project/lotus/pull/12322 .  It includes content I saw in https://github.com/filecoin-project/lotus/discussions/12020 and https://github.com/filecoin-project/lotus/issues/12072 that wasn't previously included.

### CODE REVIEW

**Comment by BigLep** [comment]
@rjan90 : I didn't want to just push these larger changes to https://github.com/filecoin-project/lotus/pull/12322 .  

This PR also doesn't include my suggestions in https://github.com/filecoin-project/lotus/pull/12322 .

I assume the best sequence of events is:
1. Incorporate/reject suggestions in https://github.com/filecoin-project/lotus/pull/12322
2. Merge whatever you like of this PR into https://github.com/filecoin-project/lotus/pull/12322
3. Work on other feedback for https://github.com/filecoin-project/lotus/pull/12322

File: LOTUS_RELEASE_FLOW.md [hunk 1]
```diff
@@ -95,12 +95,29 @@ By policy, the team will usually wait until about 3 weeks after the final releas
 
 Unless a security issue is actively being exploited or a significant number of users are unable to update to the latest version, security fixes will not be backported to previous releases.
 
+## Branch and Tag Strategy
```
Comment (BigLep) [inline LOTUS_RELEASE_FLOW.md:98]: Hmm, in re-reading https://github.com/filecoin-project/lotus/pull/12322, I'm realizing there is a lot of overlap with the "Release Process" section.  I think we could merge them.  

Maybe the heading becomes "Release Process Including Branch and Tag Strategy"?

Or we could jsut do

```
## Release Process
See https://github.com/filecoin-project/lotus/blob/master/documentation/misc/RELEASE_ISSUE_TEMPLATE.md
```

Comment (rjan90) [reply LOTUS_RELEASE_FLOW.md:98]: > Or we could just do
> ```
> ## Release Process
> See https://github.com/filecoin-project/lotus/blob/master/documentation/misc/RELEASE_ISSUE_TEMPLATE.md
> ```

I like this approach. Will incorporate it in https://github.com/filecoin-project/lotus/pull/12322 once this has merged into the `phi/update-release-flow` branch - to not cause any conflicts that needs to be resolved in this PR:

File: LOTUS_RELEASE_FLOW.md [hunk 2]
```diff
@@ -95,12 +95,29 @@ By policy, the team will usually wait until about 3 weeks after the final releas
...
+* Releases branches have the name `release/vX.Y.Z`
+* By the end of the release process, a `release/vX.Y.Z` branch will have an associated `vX.Y.Z` tag and likely also `vX.Y.Z-rcN` tags.
+* `master` branch is almost always where `release/vX.Y.Z` branches are created from.  The exception is we need to do an emergency rushed patch release and can't risk pulling in other changes that have already landed in `master`.  In that case, `release/vX.Y.Z+1` will be created from `release/vX.Y.Z`.
```
Comment (rjan90) [inline LOTUS_RELEASE_FLOW.md:101]: I think we have to think a little bit more about these lines, and how we can convey the branching strategy with the separation of `Node` and `Miner`. Ref: https://github.com/filecoin-project/lotus/pull/12096#issuecomment-2222938021

- `release/vX.Y.Z` branches for node releases
- `release/miner/vX.Y.Z` branches for miner releases
- By the end of the release process:
   - A `release/miner/vX.Y.Z` branch will have an associated `miner/vX.Y.Z` tag to it
   - A `release/vX.Y.Z` branch will have an associated `vX.Y.Z` tag to it.


Writing this in text can be a little bit confusing given the similar looking nature of these branches.

Comment (BigLep) [reply LOTUS_RELEASE_FLOW.md:101]: Good point!  I like your expanded text.  We could also do colors to help differentiate?  https://stackoverflow.com/questions/35465557/how-to-apply-color-on-text-in-markdown. Feel free to adjust/merge or I can try this out when more at the keyboard a bit later today.

Comment (rjan90) [reply LOTUS_RELEASE_FLOW.md]: Expanded the text to differentiate between node/miner, as well as colouring in: https://github.com/filecoin-project/lotus/pull/12364/commits/4d7080a689a7449678b37c5fc1dab80ef5689679

**Review by BigLep — COMMENTED** [review]
FYI that in the GitHub UI I'm not seeing colors (looking at https://github.com/filecoin-project/lotus/blob/4d7080a689a7449678b37c5fc1dab80ef5689679/LOTUS_RELEASE_FLOW.md )

**Pull request merged**